Allow for schema addition without default tree during bootstrap

KoltesDigital opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

**Description of the change**

It is currently not possible to skip default tree generation while having default schemas. While these default schemas could be added in the custom LDIF directory, they are also part of OpenLDAP, i.e. they are already embedded in the Docker image. If someday the defaults change (see #26), or less likely the files change, I'd like that they also change for me, which won't happen if I had to copy them in a own directory.

Well, the patch just inverts two nested tests.

**Benefits**

Having default schemas with a custom organization tree.

**Possible drawbacks**

Configuration with `LDAP_SKIP_DEFAULT_TREE=yes` and an empty `LDAP_CUSTOM_LDIF_DIR` had only the core schema before this patch, and will have extra schemas (cosine, inetorgperson, nis) thereafter. IMHO `LDAP_SKIP_DEFAULT_TREE=yes` should be used when `LDAP_CUSTOM_LDIF_DIR` is not empty, so the case should be rare.

**Applicable issues**

None

**Additional information**

Not important: I think this comment adds little value. https://github.com/bitnami/bitnami-docker-openldap/blob/master/2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh#L382

andresbono commented on 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:
```diff
@@ -376,13 +376,13 @@ ldap_initialize() {
-            if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
-                ldap_add_custom_ldifs
+            if is_boolean_yes "$LDAP_SKIP_DEFAULT_TREE"; then
+                info "Skipping default schemas/tree structure"
```
Should this message be changed?
```suggestion
                info "Skipping default tree structure"
```

KoltesDigital replied:
Agreed.

andresbono commented on 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:
```diff
@@ -376,13 +376,13 @@ ldap_initialize() {
-        if is_boolean_yes "$LDAP_SKIP_DEFAULT_TREE"; then
-            info "Skipping default schemas/tree structure"
+        if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
+            ldap_add_custom_ldifs
```
When entering this block of the if, should we show a message saying that we are going to skip default schemas/tree structure?

KoltesDigital replied:
Note that there's already a warning in that function, but I'm changing the patch. Overall it seems that all functions which are called from `ldap_initialize` issue their own messages, so this function only issue messages for empty blocks.

andresbono reviewed (commented):
Can you also modify the documentation for the environment variables in the README so that this new behavior is explained?

For instance, you can say that when providing `LDAP_CUSTOM_LDIF_DIR`, schemas and tree generation won't happen.

andresbono commented:
Thanks for your contribution!! Could you please check my comments?

KoltesDigital commented:
Thanks for the review! I've changed the branch (as the patch is small, I amended the commit to avoid cluttering the history).

I'm now proposing to promote "default tree" as a vocabulary term (think "set expression") in the documentation and code. Thus I've added a small section in the README to explain what this default tree is about (moving some info from the configuration section). I think this makes things a bit clearer.

andresbono commented on 2/debian-10/rootfs/opt/bitnami/scripts/libopenldap.sh:
```diff
@@ -329,8 +329,7 @@ EOF
 ldap_add_custom_ldifs() {
-    info "Loading custom LDIF files..."
-    warn "Ignoring LDAP_USERS, LDAP_PASSWORDS, LDAP_USER_DC and LDAP_GROUP environment variables..."
+    info "Loading custom LDIF files... (extra schemas won't be added, default tree won't be created)"
```
I'm not sure if this message belongs to this function. The fact that "extra schemas won't be added, default tree won't be created" depends on the if else block where `ldap_add_custom_ldifs` is called, not to this function itself...

andresbono commented:
> Thanks for the review! I've changed the branch (as the patch is small, I amended the commit to avoid cluttering the history).
> 
> I'm now proposing to promote "default tree" as a vocabulary term (think "set expression") in the documentation and code. Thus I've added a small section in the README to explain what this default tree is about (moving some info from the configuration section). I think this makes things a bit clearer.

💯  Thank you very much!
